[RHACS] ROX-19788: Adding Image scanning by using roxctl CLI documentation (#66481)
openshift/openshift-docs: cli/image-scanning-by-using-the-roxctl-cli.adoc

[id="image-scanning-by-using-the-roxctl-cli"]
:_content-type: ASSEMBLY
= Image scanning by using the roxctl CLI
include::modules/common-attributes.adoc[]
:context: image-scanning

toc::[]

[role="_abstract"]
{ocp} provides you with an integrated image registry for storing images, called the registry. By default, the registry is only accessible within the cluster where it's deployed, and each cluster can have its own registry.

The URI of an image within the registry contains information about the {ocp} project and namespace from which you can access the image. The typical URI format is `_<registry_ip>:<port>_/_<project>_/_<image>_`.

For example, an image such as `image-registry.openshift-image-registry.svc:5000/helloworld/image` is only accessible from the `helloworld` project, where `image-registry.openshift-image-registry.svc:5000` is the default hostname.

> To scan images from the registry using `roxctl` CLI, Jenkins or API, you need to ensure that each of these systems supports delegating scans to specific secured clusters. It is important that you extract the project and namespace from the image URI, and you should use project and namespace specific credentials to retrieve or scan the image.

Reviewer: The `roxctl` CLI can be used to scan images from any registry, it isn't limited to just the OCP integrated registry. 

Recommend either removing this entire section, or changing it to provide general information about scanning images with roxctl (not specific to the OCP integrated registry).

//----not sure if we need to include this content----
////

> Central now automatically guesses the namespace associated with an image based on the image path. It checks if you have access to the namespace, and then delegates the scan to a secured cluster. The namespace information is used to obtain the required secrets and access the registry. The cluster ID, `defaultClusterId` specified in the delegated registry configuration works as before.

Reviewer: This 'implementation detail' may confuse folks, my vote would be remove this paragraph.

[id="image-scanning-by-using-the-roxctl-cli"]
:_mod-docs-content-type: ASSEMBLY

> = Image scanning by using the roxctl CLI

Reviewer 2: ```suggestion
= Image scanning by by using the roxctl CLI
```
See ISG: https://www.ibm.com/docs/en/ibm-style?topic=word-usage#u

Author: I'm afraid that you might have overlooked the title. 
Current title: "Image scanning by using the roxctl CLI"
I don't think we need any more changes to the title.

include::modules/common-attributes.adoc[]
:context: image-scanning-by-using-the-roxctl-cli

toc::[]

[role="_abstract"]
You can scan images stored in image registries, including cluster local registries such as the {ocp} integrated image registry by using the `roxctl` CLI.

[id="scanning-images-in-a-remote-cluster_{context}"]
== Scanning images using a remote cluster

> By specifying the appropriate cluster in the delegated scanning configuration or the cluster parameter described in the following procedure, you can scan images from cluster local registries by using a remote cluster.

Reviewer 3: This sentence need something. It's not clear to me what is being specified where. Is either of these right?

> By specifying the appropriate cluster... in the cluster parameter described in the following procedure, 

> By specifying ... the cluster parameter as described in the following procedure....

Do we need to show how to specify the appropriate cluster in the delegated scanning configuration?

Author: > This sentence need something. It's not clear to me what is being specified where. Is either of these right?
> 
> > By specifying the appropriate cluster... in the cluster parameter described in the following procedure,
> 
> > By specifying ... the cluster parameter as described in the following procedure....
> 
Please see comment: https://github.com/openshift/openshift-docs/pull/66481#discussion_r1374957553
I think we can make a slight adjustment to the content by adding "through the cluster parameter"
"By specifying the appropriate cluster in the delegated scanning configuration or **through** the cluster parameter described in the following procedure, you can scan images from cluster local registries by using a remote cluster."
 
> Do we need to show how to specify the appropriate cluster in the delegated scanning configuration?

I have already added a  cross-ref to the delegated image scanning section.

> By specifying the appropriate cluster using the delegated scanning configuration or the cluster parameter described in the following procedure, you can scan images from cluster local registries by using a remote cluster.

Reviewer 2: ```suggestion
By specifying the appropriate cluster that uses the delegated scanning configuration or the cluster parameter described in the following procedure, you can scan images from cluster local registries by using a remote cluster.
```
See ISG: https://www.ibm.com/docs/en/ibm-style?topic=word-usage#u

Author: @dcaravel This suggestion sgtm; hope that the technical meaning does not change by incorporating this suggestion.

Reviewer: I believe it does change the technical meaning - the `cluster` is specified `in` the `delegated scanning config` - the change suggests that a cluster is being specified to `use` the `delegated scanning config`, which is not accurate.

Perhaps:

```suggestion
By specifying the appropriate cluster in the delegated scanning configuration or via the cluster parameter described in the following procedure, you can scan images from cluster local registries by using a remote cluster.
```

[IMPORTANT]
====

> For more information on how to configure delegated image scanning, see xref:../operating/examine-images-for-vulnerabilities.adoc#configuring-delegated-image-scanning_examine-images-for-vulnerabilities[Configuring delegated image scanning].

Reviewer 2: ```suggestion
For more information about how to configure delegated image scanning, see xref:../operating/examine-images-for-vulnerabilities.adoc#configuring-delegated-image-scanning_examine-images-for-vulnerabilities[Configuring delegated image scanning].
```
See ISG: https://www.ibm.com/docs/en/ibm-style?topic=word-usage#i

====

.Procedure

* Run the following command to scan the specified image in a remote cluster:
+
[source,terminal,subs="+quotes"]
----
$ roxctl image scan --image=_<image_registry>_/_<image_name>_ --cluster=_<cluster_detail>_ [flags]<1>
----

> +

Reviewer 3: Either [splitting the command over multiple lines or a definition list](https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#callouts-in-code-blocks) is the appropriate way to add callouts here

Author: @dcaravel I will be splitting command as per the doc guidelines for better readability.
$ roxctl image scan \
  --image=<image_registry>/<image_name> \// <1>
  --cluster=<cluster_detail> \// <2>
  [flags] <3>

--
<1> For `<image_registry>`, specify the registry where the image is located, for example, `image-registry.openshift-image-registry.svc:5000/`.
+
For `<image_name>`, specify the name of the image you want to scan, for example, `default/image-stream:latest`.
+

> For `<cluster_name>`, specify the name of the remote cluster, for example, `remote`.

Reviewer: I didn't catch this before, but this parameter can be the cluster `name` or cluster `id`

+
Optional: For `[flags]`, you can specify parameters to modify the behavior of the command.
+

> For more information about optional parameters, see xref:../cli/image-scanning-by-using-the-roxctl-cli.adoc#roxctl-image-scan-command-options_image-scanning-by-using-the-roxctl-cli[roxctl image scan command options].

Reviewer 3: No cross-refs in modules.

--
+
.Example output
+
[source,text,subs=attributes+]
----

> {
>   "Id": "sha256:3f439d7d71adb0a0c8e05257c091236ab00c6343bc44388d091450ff58664bf9", <1>
>   "name": { <2>
>     "registry": "image-registry.openshift-image-registry.svc:5000", <3>
>     "remote": "dave/dave-is", <4>
>     "tag": "latest", <5>
>     "fullName": "image-registry.openshift-image-registry.svc:5000/dave/dave-is:latest" <6>
>   },

Reviewer: ```suggestion
{
  "Id": "sha256:3f439d7d71adb0a0c8e05257c091236ab00c6343bc44388d091450ff58664bf9", <1>
  "name": { <2>
    "registry": "image-registry.openshift-image-registry.svc:5000", <3>
    "remote": "default/image-stream", <4>
    "tag": "latest", <5>
    "fullName": "image-registry.openshift-image-registry.svc:5000/default/image-stream:latest" <6>
  },
```

[id="image-scanning-by-using-the-roxctl-cli"]
:_content-type: ASSEMBLY
= Image scanning by using the roxctl CLI
include::modules/common-attributes.adoc[]
:context: image-scanning-by-using-the-roxctl-cli

toc::[]

[role="_abstract"]
You can use `roxctl` CLI, Jenkins, and API to scan images stored in a cluster specific {ocp} integrated image registry. To do so, specify the appropriate cluster using the delegated scanning configuration or the cluster parameter in `roxctl` CLI, Jenkins, and API.

[id="scanning-images-in-a-remote-cluster_{context}"]

> == Scanning images in a remote cluster
> 
> To scan an image in a remote cluster, use the `roxctl image scan` command. Specify the location and name of the image, and specify the remote cluster. You can also specify optional parameters to customize the behavior of the command.

Reviewer: ```suggestion
== Scanning images via a remote cluster

To scan images from cluster local registries via a remote cluster, specify the appropriate cluster using the delegated scanning configuration or use the cluster parameter described next.
```

Reviewer: Since this section is specific to the `CLI`, may be best to remove references to the API / Jenkins.

Also, may be good to link to the delegated scanning section of the docs.

.Procedure

* Run the following command to scan the specified image in a remote cluster:
+
[source,terminal,subs="+quotes"]
----
$ roxctl image scan --image=_<image_registry>_/_<image_name>_ --cluster=_<cluster_name>_ [flags]<1>
----
+
--
<1> For `<image_registry>`, specify the registry where the image is located, for example, `image-registry.openshift-image-registry.svc:5000/`.
+

> For `<image_name>`, specify the name of the image you want to scan, for example, `dave/dave-is:latest`.

Reviewer: May be use a different image name instead of `dave/dave-is:latest`, something like `default/image-stream:latest` may be easier to digest.

+
For `<cluster_name>`, specify the name of the remote cluster, for example, `remote`.
+
Optional: For `[flags]`, you can specify parameters to modify the behavior of the command.
+
For more information about optional parameters, see xref:../cli/image-scanning-by-using-the-roxctl-cli.adoc#roxctl-image-scan-command-options_image-scanning-by-using-the-roxctl-cli[roxctl image scan command options].
--
+
.Example output
+
[source,text,subs=attributes+]
----
{
  "Id": "sha256:3f439d7d71adb0a0c8e05257c091236ab00c6343bc44388d091450ff58664bf9", <1>
  "name": { <2>
    "registry": "image-registry.openshift-image-registry.svc:5000", <3>
    "remote": "dave/dave-is", <4>
    "tag": "latest", <5>
    "fullName": "image-registry.openshift-image-registry.svc:5000/dave/dave-is:latest" <6>
  },
  "names": [ <7>
    {
      "registry": "image-registry.openshift-image-registry.svc:5000",
      "remote": "dave/dave-is",
      "tag": "latest",
      "fullName": "image-registry.openshift-image-registry.svc:5000/dave/dave-is:latest"
    }

>   ],

Reviewer: Recommend removing `names` from the output and the `#7` bullet, the `names` field is more for internal purposes at this time.
